app/Http/Controllers/LinkController.php

```diff
@@ -74,12 +81,16 @@ public function performRedirect(Request $request, $short_url, $secret_key=false)
         	if (!$secret_key) {
         		// if we do not receieve a secret key
         		// when we are expecting one, return a 403
-        		return abort(403);
+                return view('error', [
+                    'message' => 'Sorry, but this link needs a secret key.'
```

Comment:
That's a good point, this could be a 404 just the same as it not existing to keep the link hidden. It could also be another env variable that lets you choose if you want to keep secret links hidden in this way?